src/librustc_resolve/macros.rs

```diff
@@ -469,4 +477,36 @@ impl<'a> Resolver<'a> {
             self.exported_macros.push(def);
         }
     }
+
+    /// Error if `ext` is a Macros 1.1 procedural macro being imported by `#[macro_use]`
+    fn err_if_macro_use_proc_macro(&mut self, name: Name, use_span: Span, binding: &NameBinding,
+                                   ext: &SyntaxExtension) {
```

Comment:
One last thing -- I hadn't considered this earlier, but the function can also be called when using a proc macro defined by a legacy plugin (`#![plugin(...)]`). In that case (and only in that case), `binding.span` will `DUMMY_SP`; I think we should add a check for that here and return in that case here.